modules/core/src/bitgo.ts

```diff
@@ -427,848 +699,920 @@ const BitGo = function(params) {
+    const secretPoint = serverDerivedNode.keyPair.__Q.multiply(clientDerivedNode.keyPair.d);
+    const secret = secretPoint.getEncoded().toString('hex');
 
-/**
- *
- * @param responseBody Response body object
- * @param password Password for the symmetric decryption
- */
-BitGo.prototype.handleTokenIssuance = function(responseBody, password) {
-  // make sure the response body contains the necessary properties
-  common.validateParams(responseBody, ['derivationPath'], ['encryptedECDHXprv']);
-
-  const environment = this.env;
-  const environmentConfig = common.Environments[environment];
-  const serverXpub = environmentConfig.serverXpub;
-  let ecdhXprv = this._ecdhXprv;
-  if (!ecdhXprv) {
-    if (!password || !responseBody.encryptedECDHXprv) {
-      throw new Error('ecdhXprv property must be set or password and encrypted encryptedECDHXprv must be provided');
-    }
+    // decrypt token with symmetric ECDH key
+    let response: TokenIssuance;
     try {
-      ecdhXprv = this.decrypt({ input: responseBody.encryptedECDHXprv, password: password });
+      response = {
+        token: this.decrypt({
+          input: responseBody.encryptedToken,
+          password: secret
+        })
+      };
     } catch (e) {
-      e.errorCode = 'ecdh_xprv_decryption_failure';
-      console.error('Failed to decrypt encryptedECDHXprv.');
+      e.errorCode = 'token_decryption_failure';
+      console.error('Failed to decrypt token.');
       throw e;
     }
+    if (!this._ecdhXprv) {
+      response.ecdhXprv = ecdhXprv;
+    }
+    return response;
   }
 
-  // construct HDNode objects for client's xprv and server's xpub
-  const clientHDNode = bitcoin.HDNode.fromBase58(ecdhXprv);
-  const serverHDNode = bitcoin.HDNode.fromBase58(serverXpub);
+  /**
+   * Calculate the string that is to be HMACed for a certain HTTP request or response
+   * @param urlPath
+   * @param text
+   * @param timestamp
+   * @param statusCode Only set for HTTP responses, leave blank for requests
+   * @returns {string}
+   */
+  calculateHMACSubject({ urlPath, text, timestamp, statusCode }: CalculateHmacSubjectOptions): string {
+    const urlDetails = url.parse(urlPath);
+    const queryPath = (urlDetails.query && urlDetails.query.length > 0) ? urlDetails.path : urlDetails.pathname;
+    if (!_.isUndefined(statusCode) && _.isInteger(statusCode) && _.isFinite(statusCode)) {
+      return [timestamp, queryPath, statusCode, text].join('|');
+    }
+    return [timestamp, queryPath, text].join('|');
+  }
 
-  // BIP32 derivation path is applied to both client and server master keys
-  const derivationPath = responseBody.derivationPath;
-  const clientDerivedNode = hdPath(clientHDNode).derive(derivationPath);
-  const serverDerivedNode = hdPath(serverHDNode).derive(derivationPath);
+  /**
+   * Calculate the HMAC for an HTTP request
+   */
+  calculateRequestHMAC({ url: urlPath, text, timestamp, token }: CalculateRequestHmacOptions): string {
```

Comment:
curious why this is ```url: urlPath``` rather than just ```url```

Reply:
Because it gets passed to `calculateHMACSubject`, which expects a param called `urlPath`, since it uses the `url` module and thus can't use that name. So something has to change the name of the property. Why it was done here instead of inside calculateHMACSubject, I'm not sure.